feat: toggle tool details visibility (resolves #4824)

ariane-emory opened the pull request:
Resolves #4824.

**Successful** tool calls can be hidden.

Unsuccessful tool calls are always displayed so that any error messages can be seen,

The modal used by the "ask" permission is NOT hidden.

In the `command_list`, the command's name is either `Hide tool usage` or `Show tool usage`, depending on its current state.

The current state of the toggle is persisted to `kv.json` as `tool_usage_visibility`.

ariane-emory committed:
feat: toggle tool action visibility, needs testing

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

ariane-emory committed:
tweak: change the command_list text to either "Hide action blocks" or "Show action blocks" depending on current state

ariane-emory committed:
...

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

ariane-emory committed:
refactor: don't add a new property openconfig.jsonc

ariane-emory committed:
Merge upstream/dev into feat/toggle-tool-action-visibility

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

rekram1-node commented on packages/opencode/src/session/message-v2.ts:
```diff
@@ -273,6 +273,7 @@ export namespace MessageV2 {
     tool: z.string(),
     state: ToolState,
     metadata: z.record(z.string(), z.any()).optional(),
+    hideDetails: z.boolean().optional(),
```
why would we need to change the schema of a message for this?? Seems  off

ariane-emory replied:
Tidies this, spurious line is no longer in HEAD.

ariane-emory committed:
tidy: revise code to eliminate the unnecessary hideDetails property.

ariane-emory committed:
tidy: revise code to eliminate the unnecessary hideDetails property.

ariane-emory committed:
tidy: revise code to eliminate the unnecessary hideDetails property.

ariane-emory committed:
fix: whoops, undo last

ariane-emory commented:
@rekram1-node Tidied up that spurious line you'd drawn my attention to. This should probably be good to go now, I think?

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

actions-user committed:
chore: format code

actions-user committed:
Update Nix flake.lock and hashes

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into feat/toggle-tool-action-visibility

ariane-emory committed:
Merge upstream/dev into feat/toggle-tool-action-visibility

rekram1-node commented on packages/opencode/src/permission/index.ts:
```diff
@@ -85,6 +92,11 @@ export namespace Permission {
     return state().pending
   }
 
+  export function wasAsked(sessionID: string, callID: string): boolean {
```
do u need this

rekram1-node commented on packages/sdk/go/session.go:
```diff
@@ -2140,6 +2141,7 @@ type toolPartJSON struct {
 	State       apijson.Field
 	Tool        apijson.Field
 	Type        apijson.Field
+	HideDetails apijson.Field
```
do u need this

rekram1-node replied:
idk why this file needed edits

rekram1-node commented on packages/ui/src/components/message-part.tsx:
```diff
@@ -130,6 +130,8 @@ PART_MAPPING["tool"] = function ToolPartDisplay(props) {
     const metadata = part.state.status === "pending" ? {} : (part.state.metadata ?? {})
     const input = part.state.status === "completed" ? part.state.input : {}
+    const shouldHide = props.hideDetails ?? false
+    const hideForError = part.state.status === "error" ? false : shouldHide
```
don't think this file needed edits?

ariane-emory committed:
refactor: remove unnecessary changes

ariane-emory commented:
@rekram1-node Seems like most of that can go. Temporarily reverting to draft status while I finish tidying it up.

ariane-emory committed:
tidy: resync this file with dev.

ariane-emory committed:
resync: update message-part.tsx with upstream/dev changes

ariane-emory committed:
refactor: remove unused wasAsked() and asked state tracking

The hiding logic only needs to check for pending permissions, not historical
tracking of tools that asked for permission. Once a tool completes successfully,
whether it required permission is irrelevant to the hiding decision.

ariane-emory committed:
fix: restore hideDetails prop for BasicTool component

- Add hideDetails to BasicToolProps interface
- Restore !props.hideDetails condition for Collapsible.Arrow visibility
- Ensures proper tool hiding functionality works correctly

ariane-emory committed:
tidy: silly Claude, changes in packages/ui/src/components/basic-tool.tsx were in fact unnecessary, resync with dev

ariane-emory committed:
Merge branch 'feat/toggle-tool-action-visibility-wip' into feat/toggle-tool-action-visibility

ariane-emory commented:
@rekram1-node Tidied up, the changes should be nice and lean now. :)

rekram1-node commented:
/oc rename the things in this pr called "tool usage" to "tool details"

opencode-agent[bot] committed:
Renamed "tool usage" to "tool details"

Co-authored-by: rekram1-node <rekram1-node@users.noreply.github.com>

rekram1-node commented:
/oc rename showActions to showDetails, also if shouldHide is true return undefined instead of null

opencode-agent[bot] committed:
Renamed showActions to showDetails, return undefined

Co-authored-by: rekram1-node <rekram1-node@users.noreply.github.com>
